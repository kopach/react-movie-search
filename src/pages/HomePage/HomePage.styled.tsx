import React from 'react'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFilm } from '@fortawesome/free-solid-svg-icons'

// TODO:
export const matrixColor = 'rgba(122, 229, 114, 0.87)'

export const HeaderCentered = styled.h1.attrs({
  className: 'text-center',
})`
  color: ${matrixColor};
  margin-top: -3rem;
`

const FilmLogo = styled(FontAwesomeIcon).attrs({
  icon: faFilm,
  color: matrixColor,
})`
  opacity: 0.3;
`

export const AppLogo: React.FC = (): JSX.Element => (
  <div className="text-center">
    <span className="fa-stack fa-7x">
      <FilmLogo className="fa-stack-1x" />
    </span>
  </div>
)

export const ContainerCentered = styled.div`
  width: 20rem;
  margin-left: auto;
  margin-right: auto;
  position: relative;
  top: 5rem;
`
