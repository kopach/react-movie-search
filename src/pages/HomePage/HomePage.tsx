import React from 'react'
import { RouteComponentProps } from 'react-router-dom'
import { SearchForm } from '../../components/SearchForm'
import { AppLogo, ContainerCentered, HeaderCentered } from './HomePage.styled'

export const HomePage: React.FC<IHomePageProps> = (): JSX.Element => {
  return (
    <ContainerCentered>
      <AppLogo />
      <HeaderCentered>Movie Search</HeaderCentered>
      <SearchForm />
    </ContainerCentered>
  )
}

interface IHomePageProps extends RouteComponentProps<{ name: string }> {}
