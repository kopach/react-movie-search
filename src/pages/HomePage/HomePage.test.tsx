import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

// @ts-ignore
import { styleSheetSerializer } from 'jest-styled-components/serializer'
// @ts-ignore
import { addSerializer } from 'jest-specific-snapshot'
import { HomePage } from './HomePage'

addSerializer(styleSheetSerializer)

const routeComponentPropsMock = {
  history: {
    location: {
      pathname: '/HomePage',
    },
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  } as any,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  location: {} as any,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  match: {} as any,
}
describe('<HomePage />', () => {
  let component: ReturnType<typeof shallow>

  beforeEach(() => {
    component = shallow(<HomePage {...routeComponentPropsMock} />)
  })

  it('should mount', () => {
    expect(component.length).toBe(1)
  })

  it('renders as expected', () => {
    const tree = renderer.create(<HomePage {...routeComponentPropsMock} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
