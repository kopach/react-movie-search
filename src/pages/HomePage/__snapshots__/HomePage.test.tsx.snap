// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders as expected 1`] = `
.c2 {
  color: rgba(122,229,114,0.87);
  margin-top: -3rem;
}

.c1 {
  opacity: 0.3;
}

.c0 {
  width: 20rem;
  margin-left: auto;
  margin-right: auto;
  position: relative;
  top: 5rem;
}

<div
  className="c0"
>
  <div
    className="text-center"
  >
    <span
      className="fa-stack fa-7x"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-film fa-w-16 c1 fa-stack-1x"
        color="rgba(122, 229, 114, 0.87)"
        data-icon="film"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M488 64h-8v20c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12V64H96v20c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12V64h-8C10.7 64 0 74.7 0 88v336c0 13.3 10.7 24 24 24h8v-20c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v20h320v-20c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v20h8c13.3 0 24-10.7 24-24V88c0-13.3-10.7-24-24-24zM96 372c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12H44c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm272 208c0 6.6-5.4 12-12 12H156c-6.6 0-12-5.4-12-12v-96c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v96zm0-168c0 6.6-5.4 12-12 12H156c-6.6 0-12-5.4-12-12v-96c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v96zm112 152c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40zm0-96c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <h1
    className="c2 text-center"
  >
    Movie Search
  </h1>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <form
        action="/search"
        autoComplete="off"
        className=""
        method="get"
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="movieTitle"
          >
            Title
          </label>
          <input
            className="form-control"
            id="movieTitle"
            name="t"
            placeholder="Title"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="movieYear"
          >
            Year
          </label>
          <input
            className="form-control"
            id="movieYear"
            name="y"
            placeholder="Year"
            type="number"
          />
        </div>
        <button
          className="btn btn-outline-primary btn-block"
          disabled={false}
          type="submit"
        >
          Search
        </button>
        <button
          className="btn btn-outline-secondary btn-block"
          disabled={true}
          type="button"
        >
          I'm Feeling Lucky
        </button>
      </form>
    </div>
  </div>
</div>
`;
